recipes/azure-storage-cpp/all/conanfile.py

```diff
@@ -0,0 +1,121 @@
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 11
+
+    @property
+    def _minimum_compiler_version(self):
+        return {
+            "gcc": "5",
+            "Visual Studio": "14",
+            "clang": "3.4",
+            "apple-clang": "5.1",
+        }
+
+    def requirements(self):
+        self.requires("cpprestsdk/2.10.18")
+        if self.settings.os != "Windows":
+            self.requires("libxml2/2.9.10")
+            self.requires("libuuid/1.0.3")
+        if self.settings.os == "Macos":
+            self.requires("libgettext/0.20.1")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        self._cmake.definitions["CMAKE_FIND_FRAMEWORK"] = "LAST"
+        self._cmake.definitions["BUILD_TESTS"] = False
+        self._cmake.definitions["BUILD_SAMPLES"] = False
+        if not self.settings.compiler.cppstd:
+            self._cmake.definitions["CMAKE_CXX_STANDARD"] = self._minimum_cpp_standard
+
+        if self.settings.os == "Macos":
+            self._cmake.definitions["GETTEXT_LIB_DIR"] = self.deps_cpp_info["libgettext"].lib_paths[0]
+
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+        min_version = self._minimum_compiler_version.get(str(self.settings.compiler))
+        if not min_version:
+            self.output.warn("{} recipe lacks information about the {} compiler support.".format(
+                self.name, self.settings.compiler))
+        else:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration("{} requires C++{} support. The current compiler {} {} does not support it.".format(
+                    self.name, self._minimum_cpp_standard, self.settings.compiler, self.settings.compiler.version))
+
+        # FIXME: Visual Studio 2015 & 2017 are supported but CI of CCI lacks several Win SDK components
+        # https://github.com/conan-io/conan-center-index/issues/4195
+        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < "16":
+            raise ConanInvalidConfiguration("Visual Studio < 2019 not yet supported in this recipe")
+
+    def build(self):
+        self._patch_sources()
```

Comment:
```suggestion
        tools.files.apply_conandata_patches(self)
```

https://docs.conan.io/en/latest/reference/conanfile/tools/files.html?conan-tools-files-apply-conandata-patches#conan-tools-files-apply-conandata-patches